src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeFieldInfo.cs

```diff
@@ -46,6 +47,12 @@ internal RuntimeType GetDeclaringTypeInternal()
 
         public override Module Module => GetRuntimeModule();
         public override bool IsCollectible => m_declaringType.IsCollectible;
+
+        public override bool Equals(object? obj) =>
+            obj == (object)this ||
+            (RuntimeTypeMetadataUpdateHandler.HotReloadDeltaApplied && CacheEquals(obj));
+
+        public override int GetHashCode() => base.GetHashCode();
```

Comment:
I do not think you can just forward to the `base.GetHashCode()`. `GetHashCode` and `Equals` have to be in sync. It means `GetHashCode` needs to have `HotReloadDeltaApplied` condition too.

Reply:
Also, add tests that verify that the hashcodes are Equal even when instances are different.

Reply:
I felt the same, but was not sure how hot reload affects GetHashCode, probably add tests to see the diff